### PR TITLE
Running websocket connection on std::thread

### PR DESCRIPTION
**Pull request by rakhad**

### CODE REVIEW

File: Source/WebSocket/Websocketpp/websocketpp_websocket.cpp [hunk 1]
```diff
@@ -606,7 +600,7 @@ struct wspp_websocket_impl : public hc_websocket_impl, public std::enable_shared
     };
 
     // Asio client has a long running "run" task that we need to provide a thread for
-    AsyncBlock* m_backgroundAsync;
+    std::thread m_websocketThread;
     async_queue_handle_t m_backgroundQueue;
```
Comment (jplafonta) [inline Source/WebSocket/Websocketpp/websocketpp_websocket.cpp]: >m_backgroundQueue [](start = 25, length = 17)

this is unused, remove?

Comment (rakhad) [reply Source/WebSocket/Websocketpp/websocketpp_websocket.cpp]: It is used when shutdown_wspp_impl is called

---
In reply to: [217564415](https://github.com/Microsoft/libHttpClient/pull/279#discussion_r217564415) [](ancestors = 217564415)

Comment (jplafonta) [reply Source/WebSocket/Websocketpp/websocketpp_websocket.cpp]: Good point, it is still needed. 

---
In reply to: [217564415](https://github.com/Microsoft/libHttpClient/pull/279#discussion_r217564415) [](ancestors = 217564415)

File: Source/WebSocket/Websocketpp/websocketpp_websocket.cpp [hunk 2]
```diff
@@ -528,7 +522,7 @@ struct wspp_websocket_impl : public hc_websocket_impl, public std::enable_shared
             auto pThis = reinterpret_cast<wspp_websocket_impl*>(async->context);
 
             // Wait for background thread to finish
-            (void)GetAsyncStatus(pThis->m_backgroundAsync, true);
+            pThis->m_websocketThread.join();
```
Comment (jplafonta) [inline Source/WebSocket/Websocketpp/websocketpp_websocket.cpp]: Just looked at the Casablanca wspp implementation and they have this:

            if (m_thread.joinable())
            {
                m_thread.join();
            }

**Pull request merged**